File: 1538-maximum-points-you-can-obtain-from-cards/1538-maximum-points-you-can-obtain-from-cards.java
class Solution {
    public int maxScore(int[] nums, int k) {

        int leftSum=0;
        int rightSum=0;
        int maxSum=0;


        for(int i=0;i<k;i++)
        {
            leftSum+=nums[i];
        }

        maxSum=leftSum;
        int right=nums.length-1;

        for(int i=k-1;i>=0;i--)
        {
           leftSum-=nums[i];
           rightSum+=nums[right];
           maxSum=Math.max(maxSum,leftSum+rightSum);
           right--;
        }

        return maxSum;
        
    }
}

File: 1538-maximum-points-you-can-obtain-from-cards/1538-maximum-points-you-can-obtain-from-cards.py
class Solution:
    def maxScore(self, cardPoints: List[int], k: int) -> int:
        total=sum(cardPoints)
        maxsum=0
        summ=0

        for i in range(0,len(cardPoints)-k):
            summ+=cardPoints[i]
        maxsum=total-summ

        for i in range((len(cardPoints) - k),len(cardPoints)):
            summ+=cardPoints[i]
            summ-=cardPoints[i-(len(cardPoints) - k)]
            maxsum=max(maxsum,total-summ)
        return maxsum
        